drivers: can: rename SocketCAN and CAN controller data types

henrikbrixandersen opened the pull request:
Rename the CAN controller API data types to not have the "z" prefix as this makes them appear as internal APIs. This requires renaming of the SocketCAN API data types to avoid naming clashes. To match the new naming, the SocketCAN data types are moved into the correct header namespace (and the corresponding tests are moved too).

I suggest reviewing this PR patch by patch to get the history correct for all the renames and reshuffling.

henrikbrixandersen committed:
drivers: can: move socketcan types and functions to socketcan header

Move the SocketCAN data types and helper functions from
zephyr/drivers/can.h to zephyr/net/socket_can.h as these are not used by
the low-level CAN controller driver API.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
tests: move the SocketCAN tests to tests/net/socket/can

Move the SocketCAN tests from tests/drivers/can/utilities to
tests/net/socket/can to match the location of the code under test.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
net: socket: can: prepend SocketCAN data types with socketcan

Rename the SocketCAN data types to "socketcan_*" in preparation of renaming
the low-level CAN controller API data types.

This breaks the naming compatibility with the similar SocketCAN data types
from the Linux kernel, but Zephyr and Linux SocketCAN are not 100%
compatible anyways (only the structure fields are compatible, extended
functionality such filtering, error reporting etc. are not).

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
drivers: can: remove z prefix from public CAN API types

Remove the "z" prefix from the public CAN controller API types as this
makes them appear as internal APIs.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
net: socket: can: rename utility functions

Rename the SocketCAN utility functions to reflect the new naming of the CAN
controller API and SocketCAN API data types.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

martinjaeger commented on drivers/can/can_mcp2515.c:
```diff
@@ -209,7 +209,7 @@ static int mcp2515_cmd_read_rx_buffer(const struct device *dev, uint8_t nm,
 }
 
-static void mcp2515_convert_zcanframe_to_mcp2515frame(const struct zcan_frame
+static void mcp2515_convert_zcanframe_to_mcp2515frame(const struct can_frame
```
I'd suggest to rename the function as well. Also the one for conversion in the other direction.

henrikbrixandersen replied:
Renamed, thanks.

martinjaeger commented on drivers/can/can_mcux_flexcan.c:
```diff
@@ -247,7 +247,7 @@ static int mcux_flexcan_set_mode(const struct device *dev, can_mode_t mode)
 }
 
-static void mcux_flexcan_copy_zframe_to_frame(const struct zcan_frame *src,
+static void mcux_flexcan_copy_zframe_to_frame(const struct can_frame *src,
```
Function could be renamed.

henrikbrixandersen replied:
Renamed, thanks.

martinjaeger commented on include/zephyr/net/socket_can.h:
```diff
@@ -51,6 +52,133 @@ struct sockaddr_can {
+/**
+ * @brief CAN frame for Linux SocketCAN compatibility.
+ */
+struct socketcan_frame {
```
I wonder if it would make sense to put this struct definition into a separate file, which can be included here and in the Linux SockedCAN "driver" I'm working on. That way we could use it as the exchange format without automatically including `zephyr/drivers/can.h`, which is required here for the conversion functions, but which has conflicting struct definitions when also the native linux can headers are included.

henrikbrixandersen replied:
Separating the two is a good idea. Either a separate header file for the `struct socketcan_*` stuff or we could move the `socketcan_*` utility functions (which require `can.h`) to a separate include file.

Which would you prefer? And what naming?

martinjaeger replied:
Moving the utility functions into a file `socketcan_utils.h` would be somewhat consistent with the `can_utils.h` in the driver (even if the latter one is a local include file only). Maybe also rename `socket_can.h` to `socketcan.h` for more consistency while we are at it?

henrikbrixandersen replied:
`socket_can.h` is named like that to be consistent with the other `socket_*.h` files, but perhaps we can change it.

martinjaeger replied:
Don't have a strong opinion here, both fine for me.

henrikbrixandersen replied:
I'll give it some thought and come up with a suggestion. Thanks.

henrikbrixandersen replied:
@martinjaeger I have renamed the header as suggested and split the utility functions into their own header.

martinjaeger reviewed (commented):
Nice clean-up. Only got some minor comments.

henrikbrixandersen committed:
drivers: can: mcp2515: rename internal funtions to not use zcan

Rename the mcp2515 driver internal functions to not use "zcan" to avoid
confusion after "struct can_frame" renaming.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
drivers: can: mcux: flexcan: rename internal funtions to not use zcan

Rename the NXP FlexCAN driver internal functions and macros to not use
"zcan", "zframe", and "zfilter" to avoid confusion after "struct can_frame"
and "struct can_filter" renaming.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
net: socketcan: rename SocketCAN header from socket_can.h to socketcan.h

Rename the SocketCAN header from socket_can.h to socketcan.h to better
match the naming of the functionality.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
net: socketcan: SocketCAN supports classic CAN, not CAN-FD

SocketCAN does not support CAN-FD data lengths of up to 64 bytes
payload. Decouple the maximum supported SocketCAN data length from the
maximum supported CAN controller data length.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
net: socketcan: fix SocketCAN MTU definition

SocketCAN transfers an entire "struct socketcan_frame", not just the
embedded payload data. Fix the MTU to reflect this.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
net: socketcan: decouple SocketCAN and CAN controller headers

Decouple the zephyr/net/socketcan.h and zephyr/drivers/can.h header files
by moving the SocketCAN utilities to their own header.

This is preparation for including the SocketCAN types defined in
socketcan.h in a native posix (Linux) SocketCAN driver context without name
clashes.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

fabiobaltieri reviewed (approved):
Cool! Do you think it would be possible to provide something to migrate an application to the new naming? Wondering if we could attach a Coccinelle patch to help convert the types.

henrikbrixandersen commented:
> Cool! Do you think it would be possible to provide something to migrate an application to the new naming? Wondering if we could attach a Coccinelle patch to help convert the types.

A simple `sed` statement will do. I could add those to the commit log, if needed.
